libraries/botframework-connector/src/auth/parameterizedBotFrameworkAuthentication.ts

```diff
@@ -0,0 +1,492 @@
+        }
+
+        const identity = await this.JwtTokenValidation_authenticateToken(
+            authHeader,
+            channelId,
+            serviceUrl
+        );
+
+        await this.JwtTokenValidation_validateClaims(identity.claims);
+
+        return identity;
+    }
+
+    private async JwtTokenValidation_validateClaims(claims: Claim[] = []): Promise<void> {
+        if (this.authConfiguration.validateClaims) {
+            // Call the validation method if defined (it should throw an exception if the validation fails)
+            await this.authConfiguration.validateClaims(claims);
+        } else if (SkillValidation.isSkillClaim(claims)) {
+            // Skill claims must be validated using AuthenticationConfiguration validateClaims
+            throw new AuthenticationError(
+                'Unauthorized Access. Request is not authorized. Skill Claims require validation.',
+                StatusCodes.UNAUTHORIZED
+            );
+        }
+    }
+
+    private async JwtTokenValidation_authenticateToken(
+        authHeader: string,
+        channelId: string,
+        serviceUrl: string
+    ): Promise<ClaimsIdentity | undefined> {
+        if (SkillValidation.isSkillToken(authHeader)) {
+            return this.SkillValidation_authenticateChannelToken(authHeader, channelId);
+        }
+
+        if (EmulatorValidation.isTokenFromEmulator(authHeader)) {
+            return this.EmulatorValidation_authenticateEmulatorToken(authHeader, channelId);
+        }
+
+        // Handle requests from BotFramework Channels
+        return this.ChannelValidation_authenticateChannelToken(authHeader, serviceUrl, channelId);
+    }
+
+    private async SkillValidation_authenticateChannelToken(
+        authHeader: string,
+        channelId: string,
+    ): Promise<ClaimsIdentity> {
+        const tokenValidationParameters: VerifyOptions = {
+            issuer: [
+                'https://sts.windows.net/d6d49420-f39b-4df7-a1dc-d59a935871db/', // Auth v3.1, 1.0 token
+                'https://login.microsoftonline.com/d6d49420-f39b-4df7-a1dc-d59a935871db/v2.0', // Auth v3.1, 2.0 token
+                'https://sts.windows.net/f8cdef31-a31e-4b4a-93e4-5f571e91255a/', // Auth v3.2, 1.0 token
+                'https://login.microsoftonline.com/f8cdef31-a31e-4b4a-93e4-5f571e91255a/v2.0', // Auth v3.2, 2.0 token
+                'https://sts.windows.net/72f988bf-86f1-41af-91ab-2d7cd011db47/', // ???
+                'https://sts.windows.net/cab8a31a-1906-4287-a0d8-4eef66b95f6e/', // US Gov Auth, 1.0 token
+                'https://login.microsoftonline.us/cab8a31a-1906-4287-a0d8-4eef66b95f6e/v2.0', // US Gov Auth, 2.0 token
+                'https://login.microsoftonline.us/f8cdef31-a31e-4b4a-93e4-5f571e91255a/', // Auth for US Gov, 1.0 token
+                'https://login.microsoftonline.us/f8cdef31-a31e-4b4a-93e4-5f571e91255a/v2.0', // Auth for US Gov, 2.0 token
+            ],
```

Comment:
Note to self: these are the same values used in the Emulator authentication flow, clean up.